src/OpenTelemetry.Instrumentation.AWSLambda/OpenTelemetry.Instrumentation.AWSLambda.csproj

```diff
@@ -1,7 +1,7 @@
 <Project Sdk="Microsoft.NET.Sdk">
 
   <PropertyGroup>
-    <TargetFramework>netstandard2.0</TargetFramework>
+    <TargetFramework>netcoreapp3.1</TargetFramework>
```

Comment:
Can we change it to net6.0? netcoreapp3.1 is deprecated and might leads to issues in CI.

Reply:
It already passed CI though (except for a timeout in the coverage build).

I wouldn't want to switch right to 6. If netcoreapp3.1 is a problem, I'd stay on netstandard2.0 and add a conditional dependency on System.Text.Json just like in https://github.com/open-telemetry/opentelemetry-dotnet-contrib/pull/1092

Reply:
> As the minimum .NET version in AWS Lambda is already 6.0, no compatibility problems are expected.

@Oberon00 Why do you not want to target net6.0, when the minimum runtime version required by ASWLambda is net6.0?

Reply:
.NET 6 is not even 2 years old. And while the .NET 6 runtime is the oldest one supported by AWS, you should still be able compile your app against older versions and run them. Also, you could use a containerized runtime with an older version.

That's why I'd prefer not to use such a recent version as minimum, especially when it's only a bugfix.

Reply:
I created PR #1273 as an alternative to this, where I do not touch the .NET version at all, staying on netstandard2.0, adding an explicit dependency on Newtonsoft.Json instead.

Probably the .NET version upgrade should be done after the next release of the package.

Reply:
.NET 6 is marked for LTS (long-term support) by .NET team and if AWS Lambda has chosen it as the minimum version, then it makes sense for its corresponding instrumentation library to do the same.

Reply:
For AWS Lambda, is it possible for a consuming application to be targeting something lower than .NET 6? If not, then why would you be bothered by any of the lower and older targets?

Reply:
@utpilla https://github.com/open-telemetry/opentelemetry-dotnet-contrib/pull/1270#discussion_r1268962025:

> you should still be able compile your app against older versions and run them. Also, you could use a containerized runtime with an older version.

To elaborate on the containerized runtime: https://docs.aws.amazon.com/lambda/latest/dg/images-create.html

I do think we should raise the requirement to net6 eventually, but not for a bugfix.

Reply:
Lets do opposite.
1. Create PR with .NET 6 bump.
2. As a follow up, fix this issue.

Reply:
If the upgrade to net6 is urgent, I'd prefer:

1. Merge a PR without a bump to net 6 (this or https://github.com/open-telemetry/opentelemetry-dotnet-contrib/pull/1273)
2. Publish a release
3. .NET 6 bump